extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSources.java

```diff
@@ -169,6 +169,10 @@ public AgroalDataSource doCreateDataSource(String dataSourceName) {
         applyNewConfiguration(dataSourceConfiguration, poolConfiguration, connectionFactoryConfiguration, driver,
                 dataSourceJdbcBuildTimeConfig, dataSourceRuntimeConfig, dataSourceJdbcRuntimeConfig, mpMetricsPresent);
 
+        if (agroalConnectionConfigurerHandle.isAvailable()) {
+            agroalConnectionConfigurerHandle.get().contributeJdbcProperties(dataSourceConfiguration);
```

Comment:
should this not be applied as soon as possible so if user explicitly set it to something else it will "win" ?

Reply:
Might not make sense for this one except if we want to allow them to shoot them in the foot but... you're right it would be better if we find other use cases.

I adjusted it!